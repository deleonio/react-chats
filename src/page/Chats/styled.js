import styled from '@emotion/styled';

export const Div_Chat = styled.div`
                                    display: flex;
                                    flex-direction: column;
                                    align-items: center;
                                    width: 310px;
                                    height: 450px;
                                    border: 10px double var(--brand-color_7);
                                    border-radius: 20px;
`
export const Div_Container = styled.div`
                                    display: flex;
                                    align-items: center;
                                    justify-content: center;
                                    flex-wrap: wrap;
`

export const Div_User = styled.div`
                                    display: flex;
                                    flex-direction: column;
                                    align-items: flex-start;
                                    border: 10px double var(--brand-color_7);
                                    border-radius: 20px;
                                    margin: 20px;
                                    width: 310px;
                                    height: 250px;
`

export const Div_Header_User = styled.div`
                                    display:flex;
                                    justify-content: center;
                                    font-family: var(--font-fam);
                                    font-size: 1.2rem;
                                    color: var(--brand-color_1);
                                    font-weight: 700;
                                    margin: 5px;
`