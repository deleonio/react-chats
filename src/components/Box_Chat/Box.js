import React from 'react'
import io from 'socket.io-client';
//Importando hooks
import {useForm} from 'react-hook-form';
//Importando components
import Input from '../Input/Input';

import Button from '../Button/Button';
//Importando styled 
import {Div_Form, Div_Input, Span_Error}  from './styled';

const Box = ({username}) => {
    const socket = io();
    /*Iniciando los hook que se van a utilizar */
    const {register, handleSubmit, errors} = useForm();

    const onSubmit = (data) => {

        if(data.chats.length !== 0){
           socket.emit('send message', {
               msg: data.chats,
               from: username
            })
          } 
      
    }

    return (
            <Div_Form>
                <form onSubmit={handleSubmit(onSubmit)}>
                    <Div_Input>
                        <Input  name="chats"  register={register} validate={{required: true, maxLength: 100, minLength: 1}} msg="Enter a Message" view_label={false}/>
                        <Button name="send" value="Send" type="submit" register={register} />
                    </Div_Input>
                </form>
                <Span_Error>
                    {errors.chats?.type === "required" && "Message is Required"}
                    {errors.chats?.type === "maxLength" && "Your Message Exceed maxLength = 100"}
               </Span_Error>
            </Div_Form>
    )
}

export default Box
