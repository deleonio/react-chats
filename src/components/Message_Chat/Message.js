import React, {useEffect, useState, useRef} from 'react';
import io from 'socket.io-client';
//Importando Componentes
import Autor from '../Autor_Chat/Autor';
//Importando Estilos
import {Div_Msg} from './styled';

const Message = ({username}) => {

    const socket = io(); 
    const [messages, setMessages] = useState([])
    /*Efecto que trae los mensajes que se han producido */
    useEffect(() => {
        socket.on('all messages', (listMsg) => {
            console.log('all messages',listMsg );
           setMessages(listMsg);
        })
        return () => {
           socket.emit('disconnected');
           socket.off();
       }
    },[])

    /*Efecto que se ejecuta cada vez que se produce un nuevo mensaje */
    useEffect(() => {
        socket.on('new message', (data) => {
            console.log('respuesta del server', data)
          setMessages([...messages, data]);
         
        })
        
        return () => {
            socket.emit('disconnected');
            socket.off();
        }
     },[messages])

     /*Efecto que se encarga de mantener constantemente subiendo el Div */
     const divRef = useRef(null);
     useEffect(() => {
         divRef.current.scrollIntoView({behavior: 'smooth'})
     })
     console.log('listado de msg', messages);

     const msg = messages.map((item, index) => {
        return (
            <Autor key={index} message={item} username={username} />
        )
    }
    )
    return (
        <Div_Msg>
            {
               msg
            }
            <div ref={divRef}></div>
        </Div_Msg>
    )
}

export default Message
