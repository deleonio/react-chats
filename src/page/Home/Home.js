import React, { useState} from 'react'
import io from 'socket.io-client';
//Importando hooks
import {useForm} from 'react-hook-form';
import { useHistory } from "react-router-dom";
//Importando components
import Input from '../../components/Input/Input';
import Button from '../../components/Button/Button';
//Importando styled
import {Div_Form, Div_Input, Span_Error}  from './styled';

const Home = () => {

     /*Iniciando los hook que se van a utilizar */
     const {register, handleSubmit, errors} = useForm();
     const history = useHistory(); //Se crea un objeto para la navegacion
     const [errorUser, setErrorUse] = useState(false);

    const onSubmit =  (data) => {
        const socket = io();
        if(data.username.length !== 0){
            socket.emit('new user', data.username, res => {
                  console.log('respuesta del cb', res)
                  res ? history.push(`/chats/${data.username}`) : setErrorUse(true);
              })
          }         
    }
    
    return (
        <Div_Form>
        <form onSubmit={handleSubmit(onSubmit)}>
                <Div_Input>
                    <Input  name="username" type="text" register={register} validate={{required: true, maxLength: 100, minLength: 1}} msg="Enter your Username" view_label={false} />
                    <Button name="send" value="Send" type="submit"/>
                </Div_Input>
                <Span_Error>
                        {errors.username?.type === "required" && "Your username is required"}
                        {errors.username?.type === "maxLength" && "Your username exceed maxLength = 100"}
                        {errorUser ? <p>Username is Active!</p> : ''}
                </Span_Error>
        </form>
        </Div_Form>
    )
}

export default Home
