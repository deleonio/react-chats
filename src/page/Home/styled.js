import styled from '@emotion/styled';

export const Div_Form = styled.div`
                                background-color: var(--brand-color_4);
                                display: flex;
                                flex-direction: column;
                                justify-content: center;
                                align-items: center;
                                width: 100%;
                                border: 5px solid var(--theme-body-border);
                                margin: 10px;
                                border-radius: 10px;
                                max-width: 280px;
                                height: 200px;
`

export const Span_Error =  styled.span`
                                display: flex;
                                flex-direction: column;
                                justify-content: center;
                                align-items: center;
                                margin: 10px;
                                font-family: var(--font-fam);
                                font-size: var(--font-size);
                                color: var(--brand-color_13);
                                font-weight: 700;
                                height: 100px;
`

export const Div_Input = styled.div`
                                display: flex;
                                justify-content: center;
                                align-items: center; 
`