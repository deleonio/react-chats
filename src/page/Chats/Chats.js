import React, {useEffect} from 'react';
import io from 'socket.io-client';
import { useParams} from "react-router-dom";
//Importando Compoenentes
import Box from '../../components/Box_Chat/Box';
import Message from '../../components/Message_Chat/Message';
import Header from '../../components/Header_Chat/Header';
//Importar Estilos
import {Div_Chat, Div_Container} from './styled';

const Chats = () => {
    const socket = io(); 

    /*Tomando el parametro enviado por la url */
    const {username} = useParams();
    
    useEffect(() => {
       socket.emit('connected', username); //Establecer conexion con el server
    },[])

    return (
        <Div_Container>
            <Div_Chat>
                <Header username={username} />
                <Message username={username} />
                <Box username={username}/>
            </Div_Chat>
        </Div_Container>
    )
}

export default Chats
