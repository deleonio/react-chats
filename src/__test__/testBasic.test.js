import React from 'react';
import { shallow} from 'enzyme';
import toJson from 'enzyme-to-json'; 


import Enzyme from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';

Enzyme.configure({ adapter: new Adapter() });

import Home from '../page/Home/Home';

it('render Home correctly', () => {
  const wrapper = shallow(
    <Home/>
  );
 
  expect(toJson(wrapper)).toMatchSnapshot();
}); 

import Chats from '../page/Chats/Chats';
import {HashRouter}  from 'react-router-dom';

it('render Chats correctly', () => {
    const wrapper = shallow(
        <HashRouter><Chats/></HashRouter> 
    );
   
    expect(toJson(wrapper)).toMatchSnapshot();
  });

import Message from '../components/Message_Chat/Message';

it('render Message correctly', () => {
    const wrapper = shallow(
        <Message />
    );
   
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  import Box from '../components/Box_Chat/Box';

  it('render Box correctly', () => {
      const wrapper = shallow(
          <Box />
      );
     
      expect(toJson(wrapper)).toMatchSnapshot();
    });


