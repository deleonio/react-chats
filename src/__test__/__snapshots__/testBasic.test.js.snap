// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Box correctly 1`] = `
<Div_Form>
  <form
    onSubmit={[Function]}
  >
    <Div_Input>
      <Input
        msg="Enter a Message"
        name="chats"
        register={[Function]}
        validate={
          Object {
            "maxLength": 250,
            "minLength": 1,
            "required": true,
          }
        }
        view_label={false}
      />
      <Button
        name="send"
        register={[Function]}
        type="submit"
        value="Send"
      />
    </Div_Input>
  </form>
  <Span_Error />
</Div_Form>
`;

exports[`render Chats correctly 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <Chats />
</Router>
`;

exports[`render Home correctly 1`] = `
<Fragment>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
  >
    <title>
      Home_Web_Chat
    </title>
    <meta
      content="home_web_chat"
      name="application-name"
    />
    <meta
      content="home_web_chat"
      name="description"
    />
    <meta
      content="notranslate"
      name="google"
    />
    <link
      href="favicon.png"
      rel="icon"
      sizes="32x32"
      type="image/png"
    />
  </HelmetWrapper>
  <Div_Form>
    <form
      onSubmit={[Function]}
    >
      <Div_Input>
        <Input
          msg="Enter your Username"
          name="username"
          register={[Function]}
          type="text"
          validate={
            Object {
              "maxLength": 100,
              "minLength": 1,
              "required": true,
            }
          }
          view_label={false}
        />
        <Button
          name="send"
          type="submit"
          value="Send"
        />
      </Div_Input>
      <Span_Error />
    </form>
  </Div_Form>
</Fragment>
`;

exports[`render Home correctly 2`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <Chats />
</Router>
`;

exports[`render Home correctly 3`] = `
<Div_Msg>
  <div />
</Div_Msg>
`;

exports[`render Message correctly 1`] = `
<Div_Msg>
  <div />
</Div_Msg>
`;

exports[`renders correctly 1`] = `
<Fragment>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
  >
    <title>
      Home_Web_Chat
    </title>
    <meta
      content="home_web_chat"
      name="application-name"
    />
    <meta
      content="home_web_chat"
      name="description"
    />
    <meta
      content="notranslate"
      name="google"
    />
    <link
      href="{/* favicon */}"
      rel="icon"
      sizes="32x32"
      type="image/png"
    />
  </HelmetWrapper>
  <Div_Form>
    <form
      onSubmit={[Function]}
    >
      <Div_Input>
        <Input
          msg="Enter your Username"
          name="username"
          register={[Function]}
          type="text"
          validate={
            Object {
              "maxLength": 100,
              "minLength": 1,
              "required": true,
            }
          }
          view_label={false}
        />
        <Button
          name="send"
          type="submit"
          value="Send"
        />
      </Div_Input>
      <Span_Error />
    </form>
  </Div_Form>
</Fragment>
`;
